tests/models/detr/test_image_processing_detr.py

```diff
@@ -159,6 +159,41 @@ def test_image_processor_from_dict_with_kwargs(self):
         self.assertEqual(image_processor.size, {"shortest_edge": 42, "longest_edge": 84})
         self.assertEqual(image_processor.do_pad, False)
 
+    def test_valid_coco_detection_annotations(self):
+        # prepare image and target
+        image = Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png")
+        with open("./tests/fixtures/tests_samples/COCO/coco_annotations.txt", "r") as f:
+            target = json.loads(f.read())
+
+        params = {"image_id": 39769, "annotations": target}
+
+        # encode them
+        image_processing = DetrImageProcessor.from_pretrained("facebook/detr-resnet-50")
+
+        # legal encodings (single image)
+        _ = image_processing(images=image, annotations=params, return_tensors="pt")
+        _ = image_processing(images=image, annotations=[params], return_tensors="pt")
+
+        # legal encodings (batch of one image)
+        _ = image_processing(images=[image], annotations=params, return_tensors="pt")
+        _ = image_processing(images=[image], annotations=[params], return_tensors="pt")
+
+        # legal encoding (batch of more than one image)
+        n = 5
+        _ = image_processing(images=[image] * n, annotations=[params] * n, return_tensors="pt")
+
+        # example of an illegal encoding (missing the 'image_id' key)
+        with self.assertRaises(ValueError) as e:
+            image_processing(images=image, annotations={"annotations": target}, return_tensors="pt")
+
+        self.assertTrue(str(e.exception).startswith("Invalid COCO detection annotations"))
```

Comment:
I see that this is to catch the `ValueError` exception

```python
raise ValueError(
    "Invalid COCO detection annotations. Annotations must a dict (single image) of list of dicts "
    "(batch of images) with the following keys: `image_id` and `annotations`, with the latter "
    "being a list of annotations in the COCO format."
)
```

I think it's good for now, but as this exception is being repeated in many parts of the library ([here](https://github.com/search?q=repo%3Ahuggingface%2Ftransformers+%22Invalid+COCO+detection+annotations%22&type=code)), maybe we can think of having a proper `COCOAnnotationError` exception.

Reply:
Indeed, there's a little duplication across scripts and a reusable custom exception would certainly be possible. 

I guess thing like the `AnnotationFormat` enum could be centralised / reused, but I wasn't sure if its duplication might be deliberate, to avoid unintended dependency headaches etc. 

If there's any appetite to see what things would look like if one sought to reduce duplication a little, I would be happy to update this PR and re-request a review.